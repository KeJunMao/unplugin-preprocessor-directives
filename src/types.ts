import type { FilterPattern } from 'vite'
import { Directive } from './core/types'

export interface Options {
  cwd: string
  directives: Directive[]
  include: FilterPattern
  exclude: FilterPattern
}

export interface UserOptions extends Partial<Options> { }
