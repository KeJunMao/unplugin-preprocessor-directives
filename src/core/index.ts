export * from './unplugin'
export * from './context'
export * from './directives'
export * from './directive'
export * from './types'
export * from './utils'
