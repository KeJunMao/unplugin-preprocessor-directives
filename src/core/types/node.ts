import type { IfToken, SimpleNode } from '.'

export interface ProgramNode extends SimpleNode {
  type: 'Program'
  body: SimpleNode[]
}

export interface CodeStatement extends SimpleNode {
  type: 'CodeStatement'
  value: string
}

export interface IfStatement extends SimpleNode {
  type: 'IfStatement'
  test: string
  consequent: SimpleNode[]
  alternate: SimpleNode[]
  kind: IfToken['type']
}

export interface DefineStatement extends SimpleNode {
  type: 'DefineStatement'
  kind: 'define' | 'undef'
  value: string
}

export interface MessageStatement extends SimpleNode {
  type: 'MessageStatement'
  kind: 'error' | 'warning' | 'info'
  value: string
}
