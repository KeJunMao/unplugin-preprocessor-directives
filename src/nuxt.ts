import type { UserOptions } from './types'
import { addVitePlugin, addWebpackPlugin, defineNuxtModule } from '@nuxt/kit'
import vite from './vite'
import webpack from './webpack'
import '@nuxt/schema'

export interface ModuleOptions extends UserOptions {

}

export default defineNuxtModule<ModuleOptions>({
  meta: {
    name: 'nuxt-unplugin-preprocessor-directives',
    configKey: 'unpluginManifest',
  },
  defaults: {
    // ...default options
  },
  setup(options, _nuxt) {
    addVitePlugin(() => vite(options))
    addWebpackPlugin(() => webpack(options))
  },
})
