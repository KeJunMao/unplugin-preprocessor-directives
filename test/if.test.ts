import { readFileSync } from 'node:fs'
import { resolve } from 'node:path'
import fg from 'fast-glob'
import { describe, expect, it } from 'vitest'
import { Context, ifDirective } from '../src'

describe('if', () => {
  const root = resolve(__dirname, './fixtures')
  const context = new Context({
    // @ts-expect-error ignore
    directives: [ifDirective],
  })

  fg.sync('if.*', { cwd: root }).forEach((file) => {
    it(`should parse ${file}, dev = true`, () => {
      context.env.DEV = true
      const code = readFileSync(resolve(root, file), 'utf-8')
      const result = context.transform(code, file)
      expect(result).toMatchSnapshot()
    })
  })

  fg.sync('if.*', { cwd: root }).forEach((file) => {
    it(`should parse ${file}, dev = false`, () => {
      context.env.DEV = false
      const code = readFileSync(resolve(root, file), 'utf-8')
      const result = context.transform(code, file)
      expect(result).toMatchSnapshot()
    })
  })
})
