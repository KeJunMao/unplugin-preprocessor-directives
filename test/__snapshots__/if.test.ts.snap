// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`if > should parse if.css, dev = false 1`] = `
"body {
}
body {
content: "!DEV";
}
body {
content: "!DEV";
}
body {
}"
`;

exports[`if > should parse if.css, dev = true 1`] = `
"body {
content: "DEV";
}
body {
content: "!DEV else";
}
body {
content: "TEST";
}
body {
content: "else";
}"
`;

exports[`if > should parse if.html, dev = false 1`] = `
"<div>!DEV</div>
<div>!DEV</div>"
`;

exports[`if > should parse if.html, dev = true 1`] = `
"<div>DEV</div>
<div>!DEV else</div>
<div>TEST</div>
<div>
<div>else</div>
</div>"
`;

exports[`if > should parse if.js, dev = false 1`] = `
"console.log('!DEV')
console.log('!DEV')"
`;

exports[`if > should parse if.js, dev = true 1`] = `
"console.log('DEV')
console.log('!DEV else')
console.log('TEST')
console.log('else')"
`;
